gateway/mw_modify_headers.go

```diff
@@ -18,8 +18,7 @@ func (t *TransformHeaders) Name() string {
 func (t *TransformHeaders) EnabledForSpec() bool {
 	for _, version := range t.Spec.VersionData.Versions {
 		if len(version.ExtendedPaths.TransformHeader) > 0 ||
-			len(version.GlobalHeaders) > 0 ||
-			len(version.GlobalHeadersRemove) > 0 {
+			(!version.GlobalHeadersDisabled && (len(version.GlobalHeaders) > 0 || len(version.GlobalHeadersRemove) > 0)) {
```

Comment:
Can we maybe start solving the negative flags by adding a `*VersionInfo.GlobalHeadersEnabled() bool` function?